src/main/resources/db/changelog/20201103112757-purge-current-sca-certs.xml

```diff
@@ -20,8 +20,29 @@
             <column name="cont_acc_cert_id"/> <!-- this sets the column value to null -->
         </update>
 
-        <delete tableName="cp_cont_access_cert" />
+        <dropForeignKeyConstraint baseTableName="cp_consumer" constraintName="fk_cont_acc_cert" />
+
+        <sql dbms="mysql,mariadb">
+            DELETE FROM cp_cont_access_cert;
+            DELETE FROM cp_owner_env_content_access;
+        </sql>
+
+        <sql dbms="postgresql,oracle,hsqldb">
+            TRUNCATE TABLE cp_cont_access_cert;
+            TRUNCATE TABLE cp_owner_env_content_access cascade;
```

Comment:
While CASCADE is safe here (as nothing has a reference into this table), we still don't want to use it because (a) it's dangerous in general (see below) and (b) it causes conflicts with the other DBMSs.

By removing the cascade, we can also remove the DBMS restriction on this block, remove the SQL block above, and allow every DB to use the TRUNCATE operation.

For clarity, when you cascade a truncate operation, it doesn't just delete the rows referencing a row in the table being truncated, *it truncates every table referencing the target table, and any tables referencing them, recursively*. You can see this in action by importing test data into a postgresql deployment, and then truncating the `cp_entitlement` table. By default, there will be zero rows in the entitlements table, and some 700+ pools in cp_pool, but by executing a TRUNCATE TABLE ... CASCADE, several other tables are truncated as well -- even though no rows are actually removed from the cp_entitlement table:

```
candlepin=> SELECT COUNT(*) FROM cp_pool;
 count 
-------
   729
(1 row)

candlepin=> SELECT COUNT(*) FROM cp_entitlement;
 count 
-------
     0
(1 row)

candlepin=> TRUNCATE TABLE cp_entitlement CASCADE;
NOTICE:  truncate cascades to table "cp_ent_certificate"
NOTICE:  truncate cascades to table "cp_pool"
NOTICE:  truncate cascades to table "cp_activationkey_pool"
NOTICE:  truncate cascades to table "cp_pool_attribute"
NOTICE:  truncate cascades to table "cp_product_pool_attribute"
NOTICE:  truncate cascades to table "cp_pool_source_stack"
NOTICE:  truncate cascades to table "cp2_pool_source_sub"
TRUNCATE TABLE

candlepin=> SELECT COUNT(*) FROM cp_pool;
 count 
-------
     0
(1 row)
```

Reply:
Agreed. Removed the cascade.